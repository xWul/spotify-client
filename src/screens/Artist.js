import React from 'react'

const Artist = (props) => {
  console.log(props)
  return <h1>AA</h1>
}

export default Artist
