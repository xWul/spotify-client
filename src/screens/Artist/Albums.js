/* LIBS */
import React, { Component } from 'react'
import styled from 'styled-components'
/* API */
import SpotifyWrapper from '../../api/index'
import { TOKEN } from '../../api/config'
const NUMBER_OF_ALBUMS = 5

const Content = styled.div`
  background-color: rgba(0, 0, 0, 0.5 );
  display: flex;
  flex-wrap: wrap;
  align-items: flex-start;
  border: 2px solid #FFF;
  position: absolute;
  width: 100%;
  height: 100%;
  opacity: 1;
  transition: all 0.3s ease;
  z-index: 110;
`

const Close = styled.span`
  cursor: pointer;
  display: block;
  padding: 10px;
  float: right;
  color: #FFF;
  width: 100%;
  text-align: right;
`
const List = styled.ul`
  width: 100%;
`
const Card = styled.li`
  display: flex;
  align-items: center;
  background-color: #FFF;
  margin: 5px;
  padding: 2px;
  border-radius: 3px;
  img {
    width: 45px;
    height: 45px;
  }
  p {
    padding: 0px 5px;
  }
`
const Load = styled.p`
  color: #FFF;
  text-align: center;
  width: 100%;
`

const Album = ({album: {name, images}}) => {
  return (
    <Card>
      <img src={images[0].url} alt={name} />
      <p>{name}</p>
    </Card>
  )
}

const AlbumsList = ({items}) => {
  return (
    <List>
      {
        items.map(album => {
          return <Album album={album} key={album.id}/>
        })
      }
    </List>
  )
}

const Loading = () => {
  return <Load>Loading...</Load>
}

class Albums extends Component {
  state = { albums: [], isLoading: false }

  componentDidMount() {
    const { id } = this.props
    this.setState({isLoading: true})
    const spotify = new SpotifyWrapper({
      token: TOKEN
    })
    spotify.artist.getAlbum(id)
      .then(response => {
        this.setState({ albums: response.items, isLoading: false })
      })
  }

  render() {
    const { onClose } = this.props
    const { albums, isLoading } = this.state
    return (
      <Content>
        <Close onClick={onClose}>Close</Close>
        {albums && <AlbumsList 
            items={albums.slice(0, NUMBER_OF_ALBUMS)} />}
        {isLoading && <Loading />}
      </Content>
    )
  }
}

export default Albums
