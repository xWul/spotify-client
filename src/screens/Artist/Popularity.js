import React from 'react'
import styled from 'styled-components'

const Label = styled.span`
  color: ${props => props.color ? props.color : '#333'};
  font-weight:  bold;
`

const Content = styled.p` 
  width: 100%;
`

const Popularity = ({ popularity }) => {
  let label = <Label>Underground</Label>
  if (popularity >= 80) {
    label = <Label color={'#FB5656'}>Hot</Label>
  } else if (popularity >= 60) {
    label = <Label color={'#EAEA00;'}>Cool</Label>
  } else if (popularity >= 30) {
    label = <Label color={'#7E7EE3'}>Regular</Label>
  }

  return <Content>{label}</Content>
}

export default Popularity
