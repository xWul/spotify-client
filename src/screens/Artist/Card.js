/* LIBS */
import React, { Component } from 'react'
import styled from 'styled-components'
import { screens } from '../../theme/globalStyle'
import { setFavorite, removeFavorite, getFavorites } from '../../utils'
/* COMPONENTS */
import LikeButton from '../../components/LikeButton'
import Popularity from './Popularity'
import Albums from './Albums'

const Card = styled.div`
  font-family: 'Montserrat', sans-serif;
  position: relative;
  background-color: #FFF;
  color: #333;
  flex-wrap: wrap;
  display: flex;
  margin: 1.5%;
  width: 30%;
  @media (max-width: ${screens.medium}) {
    width: 47%;
  }
  @media (max-width: ${screens.small}) {
    width: 100%;
    margin: 2% 0px;
  }
`
const Picture = styled.div`
  width: 100%;
  height: 200px;
  background: ${props => props.images[0] ? `url(${props.images[0].url})` : '#CCC'};
  background-position: center;
  background-size: cover;
  cursor: pointer;
`
const Title = styled.h3`
  font-size: 20px;
  padding: 7px 0; 
`

const Header = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
`

const Genders = styled.p` 
  font-size: 14px;
  color: #666;
  width: 100%;
  padding: 5px 0; 
`

const CardInfo = styled.div`
  padding: 10px; 
  font-size: 14px;
  width: 100%;
  min-height: 140px;
`

const genresList = (items) => {
  return items.map(genre => genre).join(', ')
}

class ArtistCard extends Component {
  state = { 
    isFavorite: false,
    showAlbums: false 
  }

  componentDidMount() {
    const { id } = this.props.artist
    const favorites = getFavorites()['artist'] || []
    if (favorites.includes(id)) {
      this.setState({isFavorite: true})
    }
  }

  toggleFavorite = (id) => {
    this.setState(({ isFavorite }) => ({
      isFavorite: !isFavorite
    }), () => {
      const {isFavorite} = this.state
      isFavorite 
        ? setFavorite('artist', id) 
        : removeFavorite('artist', id)
    })
  }

  toggleAlbums = () => {
    this.setState(({ showAlbums }) => ({
      showAlbums: !showAlbums
    }))
  }

  render() {
    const { id, name, images, genres, popularity } = this.props.artist
    const { isFavorite, showAlbums } = this.state
    return (
      <Card>
        {showAlbums && <Albums id={id} onClose={this.toggleAlbums} />}
        <Picture images={images} onClick={this.toggleAlbums} />
        <CardInfo>
          <Header>
            <Title>{name}</Title>
            <LikeButton
              onClick={() => this.toggleFavorite(id)}
              isChecked={isFavorite}/>
          </Header>
          <Popularity popularity={popularity} />
          <Genders>{genresList(genres)}</Genders>
        </CardInfo>
      </Card>
    )
  }
}

export default ArtistCard
