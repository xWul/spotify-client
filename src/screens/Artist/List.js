/* LIBS */
import React from 'react'
import styled from 'styled-components'
/* COMPONENTS */
import Card from './Card'

const Container = styled.div`
  width: 100%;
  display: flex;
  flex-wrap: wrap;
`

const ArtistList = ({ items = [] }) => {
  return (
    <Container>
      {
        items.map(item => {
          return <Card key={item.id} artist={item} />
        })
      }
    </Container>
  )
}

export default ArtistList
