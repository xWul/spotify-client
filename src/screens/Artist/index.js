/* LIBS*/
import React, { Component } from 'react'
import styled from 'styled-components'
import { withRouter } from 'react-router-dom'
import { screens } from '../../theme/globalStyle'
/* API */
import SpotifyWrapper from '../../api/index'
import { TOKEN } from '../../api/config'
/* COMPONENTS */
import Search from '../../components/Search'
import Loader from '../../components/Loader'
import Alert from '../../components/Alert'
import List from './List'

const Container = styled.main`
  min-height: 100vh;
  background-color: #1D1D1D;
  padding: 40px 10%;
  @media(max-width: ${screens.small}) {
    padding: 40px 5% 100px 5%;
  }
`

class Artist extends Component {
  state = {
    artists: [],
    isLoading: false,
    errorMessage: false
  }

  componentDidMount() {
    const { query } = this.props.match.params
    this.fetchArtist(query)
  }

  componentWillReceiveProps(props) {
    const { query } = props.match.params
    this.fetchArtist(query)
  }

  fetchArtist = (query) => {
    this.setState({isLoading: true})
    const spotify = new SpotifyWrapper({
      token: TOKEN
    })
    spotify.search.artists(query)
    .then(data => {
      this.setState({isLoading: false})
      if (data.error) {
        this.setState({errorMessage: data.error.message})
      }else {
        this.setState({artists: data.artists.items})
      }
    })
  }

  render() {
    const { query } = this.props.match.params
    const { artists, isLoading, errorMessage } = this.state
    return (
      <Container>
        {isLoading && <Loader/> }
        <Search option={'artist'} query={query} />
        {errorMessage && <Alert>{errorMessage}</Alert>}
        <List items={artists} />
      </Container>
    )
  }
}


export default withRouter(Artist)
