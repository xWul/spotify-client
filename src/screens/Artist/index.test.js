import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import mockFetch from '../../api/mockFetch'
import Artist from './index'

describe('Artist', function () {
  test('Artists renderer without crashing', () => {
    fetch = mockFetch({artists: {items: []}})
    const component = renderer.create(
      <Router><Artist /></Router>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
