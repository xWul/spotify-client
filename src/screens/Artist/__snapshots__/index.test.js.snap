// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist Artists renderer without crashing 1`] = `
<main
  className="sc-kkGfuU khwjdg"
>
  <div
    className="sc-iwsKbI lhnNTF"
  >
    <div
      className="sc-gZMcBi bLjqeM"
    >
      <div
        className="sc-gqjmRU jraKAo"
      />
    </div>
  </div>
  <div
    className="sc-dnqmqq haujyR"
  >
    <form
      className="sc-gzVnrw cAeJnK"
      method="POST"
      onSubmit={[Function]}
    >
      <div
        className="sc-htpNat farngo"
      >
        <div
          className="sc-bdVaJa LDKxg"
        >
          <select
            className="sc-bwzfXH ljcehE"
            name="option"
            onChange={[Function]}
            selected="artist"
            value="artist"
          >
            <option
              value="artist"
            >
              Artists
            </option>
            <option
              value="album"
            >
              Albums
            </option>
            <option
              value="track"
            >
              Tracks
            </option>
          </select>
        </div>
      </div>
      <div
        className="sc-bxivhb kBJQUN"
      >
        <input
          className="sc-ifAKCX cJifCe"
          name="query"
          onChange={[Function]}
          placeholder="Search for Artists, Albums or Tracks..."
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-EHOje dmZUow"
      >
        <button
          className="sc-bZQynM cYeeSS"
          type="submit"
        >
          Search
        </button>
      </div>
    </form>
  </div>
  <div
    className="sc-kEYyzF dpgOga"
  />
</main>
`;
